### DISCUSSION TITLE
[ANNOUNCEMENT] Pulsar Admin Go Library introduction

### DISCUSSION THREAD

**Discussion by ericsyh** [apache/pulsar#19932] [Show and tell]
Hi Community, 

We want to announce and introduce the [Pulsar Admin Go Library](https://github.com/streamnative/pulsar-admin-go) which has been open sourced and cut the first release. 

The original Pulsar Admin Go Library is from [pulsarctl](https://github.com/streamnative/pulsarctl)'s internal pkg which is not convenient for Pulsar Golang developers to use and manage. So, we decoupled the Pulsar Admin related API from pulsarctl and created the [pulsar-admin-go](https://github.com/streamnative/pulsar-admin-go) library based on it, which also provides a clearer perspective and maintainability from an architectural perspective.

In the future plan of this library:
- Iterate the library and upgrade the related admin API based on the Pulsar progress.
- More testing and dogfood from StreamNative’s golang projects: [pulsarctl](https://github.com/streamnative/pulsarctl), [terraform-provider-pulsar](https://github.com/streamnative/terraform-provider-pulsar), [pulsar-resources-operator](https://github.com/streamnative/pulsar-resources-operator).
- When the library is stable, will donate it to Apache Foundation.

**Comment by tisonkun** [comment]
Sounds interesting :)

**Reply by flowchartsman** [reply]
Great news! This will make my job much easier, as I've got a couple projects that make relatively heavy use of the pulsarctl internals for exactly this reason.  This is also great timing, since once #20443 lands, Go functions will have access to the web service URL, and then it would be that much easier to achieve feature Parity with the Java SDK.

**Reply by tisonkun** [reply]
Good to know!

I've reviewed https://github.com/apache/pulsar/pull/20443 and there still some issues to investiagte. I'm glad to follow your works now :D
